src/Data/Managers/V2/GETTERS/GetInstanceWithRelations/useGetInstanceWithRelations.ts

```diff
@@ -60,6 +68,97 @@ export const useGetInstanceWithRelations = (
     return await response.json();
   };
 
+  /**
+   * This function is responsible for extracting the names of all inter-service relations from the provided service model or embedded entity.
+   * It also recursively extracts the names of all relations from any embedded entities within the provided service model or embedded entity.
+   *
+   * @param {ServiceModel | EmbeddedEntity} serviceModel - The service model or embedded entity from which to extract the relations.
+   * @returns {string[]} An array of the names of all relations.
+   */
+  const getAllRelationsNames = (
+    serviceModel: ServiceModel | EmbeddedEntity,
+  ): string[] => {
+    const relations =
+      serviceModel.inter_service_relations?.map((relation) => relation.name) ||
+      [];
+
+    const nestedRelations = serviceModel.embedded_entities?.flatMap((entity) =>
```

Comment:
Here I should remove "?" because according to interface embedded entities in service model cant be undefined, I checked on test env it seems to align